plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoColumnHandle.java

```diff
@@ -35,9 +41,26 @@ public record MongoColumnHandle(String baseName, List<String> dereferenceNames,
     public MongoColumnHandle
     {
         requireNonNull(baseName, "baseName is null");
-        dereferenceNames = ImmutableList.copyOf(requireNonNull(dereferenceNames, "dereferenceNames is null"));
+        requireNonNull(dereferenceNames, "dereferenceNames is null");
         requireNonNull(type, "type is null");
         requireNonNull(comment, "comment is null");
+
+        if (dbRefField) {
+            String leafColumnName = dereferenceNames.getLast();
+            String leafDBRefNativeName = switch (leafColumnName) {
+                case DATABASE_NAME -> DATABASE_NAME_NATIVE;
+                case COLLECTION_NAME -> COLLECTION_NAME_NATIVE;
+                case ID -> ID_NATIVE;
```

Comment:
I think I remember why we disallowed DBRef. We can't distinguish DBRef from object with same field names. 
This PR causes silent correctness issue if the document has `databaseName`, `collectionName` and `id` fields.

Reply:
@ebyhr I don't think that's the case - I [added a test](https://github.com/trinodb/trino/pull/22027/commits/cbfcde0d7c30fc7b83cd1cd41b5a4969389fdb79) to check this scenario.

The only time this would be a problem is if the same collection has documents that hold both actual DBRefs and DBRef-like documents in the same field.

Could we just document this possible but extremely unlikely edge case? IMO the importance of DBRef pushdown is way higher than this concern since without it you're loading the entire collection into memory when filtering or joining by a DBRef field (which makes Trino fairly useless for bigger databases in such cases).

Reply:
The added test is wrong. Please reorder `creator` fields as databaseName, collectionName, id.

Reply:
You're right - I put them in the order that mongo expects them ([docs](https://www.mongodb.com/docs/manual/reference/database-references/#format)), not the order trino expects them. If I order them as you said the test does indeed fail.

Is there a way we could signal that a field is a DBRef in its type signature so that we didn't have to rely on field names & order to detect a DBRef down the line?

Reply:
@ebyhr in order to consider DBRef pushdown do you think we should
a) find a way to support it cleanly (how? a custom type?)
b) require it to be explicitly turned on with a disclaimer that it can break on specifically formed documents?

Reply:
ping @ebyhr - would appreciate some guidance

Reply:
@ebyhr What if we could allow the pushdown - if the Collection doesn't have any columns on those name ?

Reply:
@Praveen2112 I think the problem is we can't really differentiate whether those are columns on the collection or these "system" DBRef columns.

Reply:
see @ebyhr  's first comment in this thread.